ethereum/geth_client.go

```diff
@@ -373,59 +382,17 @@ type callContexter interface {
 }
 
 // incrementNonce updates the opts.Nonce to next valid nonce
-// We pick the current nonce by getting latest transactions included in the blocks including pending blocks
-// then we check the txpool to see if there any new transactions from the address that are not included in any block
-// If there are no pending transactions in txpool, we use the current nonce + 1
-// else we increment the greater of current nonce or the nonce derived from txpool
-func (gc *gethClient) incrementNonce(opts *bind.TransactOpts, txpoolAccessEnabled bool, noncer noncer, cc callContexter) error {
-	// check if the txpool access is enabled
-	if !txpoolAccessEnabled {
-		log.Warningf("Ethereum Client doesn't support txpool API, may cause transactions failures")
-		opts.Nonce = nil
-		return nil
-	}
-
+func (gc *gethClient) incrementNonce(opts *bind.TransactOpts) error {
```

Comment:
I would rename it to:
```suggestion
func (gc *gethClient) setNonce(opts *bind.TransactOpts) error {
```